README
FEM (Finite Element Method) Code

Compiling
=========

Solver: cd fem ; make

User Interface:  cd tui ; make all gfem


Notes:
======

Solver (fem) includes its own documentation it its directory
(see README, TODO, PLATFORMS etc.).

Solver is fully independent on user the interface and is distributed
under GNU General Public Licence, version 2.

Most parts of the user interface are 
under Lesser GNU General Public Licence, version 2 
(the rest is under GPL).


ChangeLog
Changelog for the user interface only.

See fem/ChangeLog for solver changes.



### Fri Jan 26 20:04:37 CET 2018

version 0.2.80

features:
  + suppoort for plastic strain (data fileds only) in e002, e009, e010, e012



### Wed Nov  1 20:37:24 CET 2017

version 0.2.79

bugfixes:
  - segfault in APDL export



### Sun Sep 17 21:34:24 CEST 2017

version 0.2.78

bugfixes:
  - aextrude fix: ignore unselected areas



### Mon Aug 28 22:17:19 CEST 2017

version 0.2.77

bugfixes:
  - fdb_user_get_first_selected always returned 1st member



### Thu Aug  3 11:04:49 CEST 2017

version 0.2.76

bugfixes:
  - if one solver thread is specified then all pthreads code is ommited



### Thu Dec  8 20:11:01 CET 2016

version 0.2.75

features:
  + date can be hidden from plot (plotprop,date,no)

### Sun Oct 30 15:31:23 CET 2016

version 0.2.74

bugfixes:
  - meshing of cvbrick (g.e. type 004) fixed - it's much 100+ faster now



### Sat Oct 22 21:25:27 CEST 2016

version 0.2.73

features:
  + brick meshing speedup (a little one)

bugfixes:
  - material 010 fixed (full sigma is used now)


### Wed Aug 24 09:12:33 CEST 2016

version 0.2.71

features:
  + gui selection to define a path by 2 nodes

bugfixes:
  - invalid dialog for the "ptwo" command



### Tue Aug 23 20:17:54 CEST 2016

version 0.2.71

features:
  + possibility to define a path by 2 nodes



### Tue Aug 16 19:43:28 CEST 2016

version 0.2.69

bugfixes:
  - failing interpreter on ARM devices (memory leak)
  - string handling bug (fail on empty parameters)



### Sun Aug  7 20:50:15 CEST 2016

version 0.2.68

features:
  + "plns,u_sum" and "prns,u_sum" implemented

bugfixes:
  - FEM export fixed (uninitialized parameter)



### Thu Apr 14 19:06:08 CEST 2016

bugfixes:
  - segfault in "?sel,," functions with empty arguments


 
### Mon Jan  4 19:10:25 CET 2016

fixes:
  + segfault in fdbExport for export_to_mac (invalid params)



### Sun Nov 22 19:37:18 CET 2015

fixes:
  + tollerance test in *select (precision slightly lowered)



### Tue Aug 11 20:31:36 CEST 2015

fixes:
  + fixed node results on 8-node brick



### Mon Aug 11 17:36:56 CEST 2014

features:
  + menu item for fast reading of single results
  + command rrread for fast readinge of single results

### Mon Aug  4 21:07:09 CEST 2014

features:
  + *extrude may use MAT properties from the original area



### Wed Jul 23 20:43:19 CEST 2014

version 0.2.66

features:
  + CSV output for *list commands
  


### Tue Jul  9 21:53:46 CEST 2013

version 0.2.65

features:
  + triangle elements (11, 18) can be used for automatic regular meshes



### Thu Jan 31 19:11:00 CET 2013

version 0.2.64

features:
  + modal analysis from GUI



### Sun Nov  4 22:07:23 CET 2012

features:
  + user-defined GUI dialogs



### Wed Oct 17 21:05:42 CEST 2012

features:
  + symbols for heats (just a simple dots)


  
### Sat Oct 13 20:34:39 CEST 2012

version 0.2.63

features:
  + updated material 001



### Tue Jun 19 22:37:17 CEST 2012

features:
  + fixed maximal deformation (for animations)
  + fixed color palette (for animations)



### Mon Jun 11 22:13:11 CEST 2012

bugfixes:
  - invalid entities for "gekgen" command



### Sun Jun  3 20:38:20 CEST 2012

features:
  + extrude function for areas and lines
  + enhanced apdl export

bugfixes:
  - iterations number not passed to  solver



### Fri Feb 17 20:48:17 CET 2012

features:
  + coordinate axes symbol in [0,0,0]



### Tue Feb  7 20:44:11 CET 2012

features:
  + aextrude command 

bugfixes:
  - extrude for 4-node areas fixed
  - extrude for 8-node areas fixed



### Sun Feb  5 18:36:35 CET 2012

version 0.2.62

features:
  + new geometry - curvilinear rectangle



### Sun Jan 15 18:59:57 CET 2012

version 0.2.61

features:
  + plotting of deformations (no label, ux,uy,uz only)

bugfixes:
  - Czech documentation
  - missing item label in plns



### Fri Jan  6 21:03:55 CET 2012

version 0.2.58

features:
  + import of tetrahedra mesh from TetGen (tetgen.org)

bugfixes:
  - bad string in "import" command



### Mon Jan  2 19:32:04 CET 2012

version 0.2.58

features:
  + <Enter>, <Esc> can be used in dialogs for easier use



### Sun Nov  6 18:59:39 CET 2011

version 0.2.58

features:
  + two more parameters for material 004 (non-compatible data change)
  + GUI: RS types checked against existing ET in creation dialog



### Mon Oct 17 23:32:20 CEST 2011

features:
  + GUI items for path operations (unstested)



### Thu Aug 11 16:00:26 CEST 2011

version 0.2.57

features:
  + assesment routine for steel links
  + input data for e001, e007 changed (for assesment)



### Mon Jun 27 20:23:05 CEST 2011

version 0.2.56

features:
  + thermal element 019 (tetrahedron)



### Sat Jun 25 20:12:13 CEST 2011

version 0.2.55

features:
  + stationary heat transfer problem (2D only)
  + transfer of temperatures to plane stress model



### Sun Apr 10 20:20:42 CEST 2011

version 0.2.54

features:
  + change of m001 definition



### Thu Jan 13 13:44:18 CET 2011

bugfixes:
  - ininitialized values in data structure (now set to NULL)



### Thu Dec 16 14:33:17 CET 2010

features:
  + multiple paths plotting (untested)



### Mon Nov 29 14:40:35 CET 2010

features:
  + damping parameters  added as random variables



### Mon Nov 29 13:10:31 CET 2010

features:
  + -ssor parameter added to all solver commands



### Tue Nov 16 22:12:33 CET 2010

bugfixes:
  - path filed size problems
  - faults on 64bit Linux fixed (problem related to paths)



### Tue Nov  9 20:05:14 CET 2010

features:
  + fem export adds 0 for dynamics (for compatibility)



### Sun Oct 31 19:26:38 CET 2010

bugfixes:
  - plotting of pathes > 1 was refused



### Sun Oct 31 12:40:11 CET 2010

bugfixes:
  - invalid line width after path plotting



### Sat Oct 23 16:51:27 CEST 2010

features:
  + path picking ("gcreate,pick"_



### Fri Oct 22 19:39:13 CEST 2010

features:
  - zoom for plotted paths

bugfixes:
  - Makefiles (invalid ifeq condition)

fdb/fdb.c
/*
   File name: fdb.c
   Date:      2003/06/16 22:24
   Author:    Jiri Brozovsky

   Copyright (C) 2003 Jiri Brozovsky

   This library is free software; you can redistribute it and/or
   modify it under the terms of the GNU Library General Public
   License as published by the Free Software Foundation; either
   version 2 of the License, or (at your option) any later version.
  
   This library is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   Library General Public License for more details.
  
   You should have received a copy of the GNU Library General Public
   License along with this library; if not, write to the Free Software
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
   USA.

	 Database for FEM

	 $Id: fdb.c,v 1.1 2003/08/02 18:43:33 jirka Exp $
*/

#include "fdb.h"

int femCmdNumProcUI = 1 ;  /* number of processes for operations */

/* end of fdb.c */
